feat(menu,base): annotate mdc-menu for closure compiler (#339)

bogusred opened the pull request:
Added new annotations to mdc-menu. Corrected a couple of annotations in base.

bogusred commented:
I'm a Googler so I believe I'm not required to sign. But let me know if I misunderstood.

FYI Karma is failing in master for me on my local machine. Let's discuss next week. FYI I'm OOO Monday.

```
ERROR in ./packages/mdc-textfield/index.js
Module not found: Error: Can't resolve '@material/ripple' in '/Users/ssahim/gitprojects/mdc-web/packages/mdc-textfield'
 @ ./packages/mdc-textfield/index.js 36:53-98
 @ multi ./packages/mdc-textfield/index.js
16 06 2017 13:50:40.567:INFO [karma]: Karma v1.7.0 server started at http://0.0.0.0:9876/
16 06 2017 13:50:40.569:INFO [launcher]: Launching browser Chrome with unlimited concurrency
16 06 2017 13:50:40.574:INFO [launcher]: Starting browser Chrome
16 06 2017 13:50:41.306:INFO [Chrome 58.0.3029 (Mac OS X 10.12.5)]: Connected on socket ArzG3ehf2womb1taAAAA with id 92725245
16 06 2017 13:50:41.997:WARN [reporter]: SourceMap position not found for trace: undefined
16 06 2017 13:50:41.998:WARN [reporter]: SourceMap position not found for trace: undefined
Chrome 58.0.3029 (Mac OS X 10.12.5) ERROR
  Uncaught Error: Cannot find module "@material/ripple"
  at test/unit/index.js:3978
Chrome 58.0.3029 (Mac OS X 10.12.5): Executed 0 of 0 ERROR (0.696 secs / 0 secs)

```

bogusred commented:
Looks like I need to change the commit email to be from my google.com email address. I'll make that update next week.
FYI I've just registered my github account with google.

bogusred commented:
Looks like some tests failed in travis including lint errors. i'll address these next week.

lynnjepsen commented on packages/mdc-menu/simple/adapter.js:
```diff
@@ -0,0 +1,127 @@
+/**
+ * Provides type information for the adapter to the Closure compiler.
```
This documentation should say more about what SimpleMenu is, and what it's adapter has to handle.

bogusred commented:
Thank you Lynn. `npm i` fixed local karma. \o/
I've also updated the adapter with:
- google copyright
- fix lint errors
- add docs about adapter. let me know if this was what you were going for or if you have an example you can point me to.

Also I am a n00b at pull requests and am not sure of the best workflow to send updates. Looks like the previous snapshot of the code was lost (probably because i rebased and force pushed?). Sent you a separate email about that. If you have any workflow tips I'd very much appreciate.

Looks like something else in the CI is broken so i'll take a look and fix soon.

lynnjepsen commented:
Travis CI is failing for reasons unrelated to your change. We've got a fix in the works...once its submitted ill update this branch and that will kick off Travis again. Hopefully that gets all the tests to pass :D

Docs look good, very thorough. Thank you!

lynnjepsen commented:
Codecov is being silly, I ran this locally and there is no change in coverage. Thanks again Susie! I'm going to merge this now.

bogusred commented:
yay! \o/ Thank you Lynn
